src/rebar_packages.erl

```diff
@@ -19,27 +20,44 @@
 get_packages(State) ->
     RegistryDir = package_dir(State),
     DictFile = filename:join(RegistryDir, "dict"),
+
+    case filelib:is_file(DictFile) of
+        true ->
+            try
+                {ok, DictBinary} = file:read_file(DictFile),
+                binary_to_term(DictBinary)
+            catch
+                _:_ ->
+                    ?ERROR("Bad packages index, try to fix with `rebar3 update`", []),
+                    dict:new()
+            end;
+        false ->
+            ?ERROR("Bad packages index, try to fix with `rebar3 update`", []),
+            dict:new()
+    end.
```

Comment:
I think the entire `case ... of` here is unnecessary given your usage of `try...catch` on file handling. Is the Dict thing new anyway or I just missed it before?